File: sdk-python/examples/forms/add_logic_rule.py
"""Run with: python examples/forms/add_logic_rule.py

Conditional logic reveals a target element based on a source element's answer.
Selection questions match on which options are chosen
(``SelectionLogicCondition``); Yes/No questions match on the answer value
(``YesNoLogicCondition``).
"""

from __future__ import annotations

from threecommon import ThreeCommon
from threecommon.forms import AddLogicRuleBody, SelectionLogicCondition, YesNoLogicCondition


def main() -> None:
    with ThreeCommon(api_key="3co_your_api_key_here") as client:
        # Selection question: reveal the target when the first option is chosen.
        element = client.forms.add_logic_rule(
            "frm_replace_with_real_id",
            "elm_select_id",
            AddLogicRuleBody(
                revealed_element_id="elm_revealed_id",
                condition=SelectionLogicCondition(option_indices=[0], operator="any_of"),
            ),
        )
        print(f"added selection rule on element {element.id} ({element.type})")

        # Yes/No question: reveal the target when the respondent answers "yes".
        element = client.forms.add_logic_rule(
            "frm_replace_with_real_id",
            "elm_yes_no_id",
            AddLogicRuleBody(
                revealed_element_id="elm_other_revealed_id",
                condition=YesNoLogicCondition(selection_type="is", value=True),
            ),
        )
        print(f"added Yes/No rule on element {element.id} ({element.type})")


if __name__ == "__main__":
    main()

File: sdk-node/examples/forms/add-logic-rule.ts
/**
 * Add conditional-logic rules. Selection questions reveal a target element
 * based on which options are chosen; Yes/No questions reveal a target element
 * based on the answer value.
 *
 * Run:
 *   npx tsx examples/forms/add-logic-rule.ts
 */

import { ThreeCommon } from '@3common/sdk'

const client = new ThreeCommon({
  apiKey: '3co_your_api_key_here',
})

// Selection question: reveal the target when the first option is chosen.
const selectSource = await client.forms.addLogicRule('frm_replace_with_real_id', 'elm_select_id', {
  revealedElementId: 'elm_target_id',
  condition: { optionIndices: [0], operator: 'any_of' },
})

console.log(`added selection rule to element ${String(selectSource.id)} (${selectSource.type})`)

// Yes/No question: reveal the target when the respondent answers "yes".
const yesNoSource = await client.forms.addLogicRule('frm_replace_with_real_id', 'elm_yes_no_id', {
  revealedElementId: 'elm_other_target_id',
  condition: { selectionType: 'is', value: true },
})

console.log(`added Yes/No rule to element ${String(yesNoSource.id)} (${yesNoSource.type})`)
